src/pages/fareType.tsx

```diff
@@ -98,32 +103,41 @@ export const getServerSideProps = async (ctx: NextPageContextWithSession): Promi
     const csrfToken = getCsrfToken(ctx);
 
     const schemeOp = isSchemeOperator(ctx);
-    const opIdentifier = getAndValidateNoc(ctx);
+    const nocCode = getAndValidateNoc(ctx);
 
-    const services = await getServicesByNocCode(opIdentifier);
+    const bodsServices = await getServicesByNocCode(nocCode, 'bods');
```

Comment:
Just thinking about this, if we go down that route then we will be removing the getServicesByNocCodeAndDataSource function altogether and instead using filtering on the array we get back every time? It seems a lot of fiddling around for not much gain unless I'm misunderstanding something?

Reply:
So we can keep that function and we could just use the new `getAllServicesByNocCode` function here and then just check if it contains bods / tnds data with some js filters. Generally if you can save doing a db query then you should since they're the one of the more expensive things to do. So we could have something like:
```
const services = await getAllServicesByNocCode(nocCode);
    const hasBodsServices = services.some(service => service.dataSource === 'bods');
    const hasTndsServices = services.some(service => service.dataSource === 'tnds');

    if (!schemeOp && services.length === 0) {
        if (ctx.res) {
            redirectTo(ctx.res, '/noServices');
        } else {
            throw new Error(`No services found for NOC Code: ${nocCode}`);
        }
    }
    const dataSouceAttribute = getSessionAttribute(ctx.req, TXC_SOURCE_ATTRIBUTE);
    if (!dataSouceAttribute) {
        updateSessionAttribute(ctx.req, TXC_SOURCE_ATTRIBUTE, {
            source: hasBodsServices && !hasTndsServices ? 'bods' : 'tnds',
            hasBods: hasBodsServices,
            hasTnds: hasTndsServices,
        });
    }
```